server/routes/listingRouter.js

```diff
@@ -1,45 +1,76 @@
 const listingRouter = require('express').Router();
-// const { Listing } = require('../models.js'); <- uncomment when Listing model is implemented
+const { Listing } = require('../models.js');
 const { passport } = require('../jwtEncrypt.js');
 
-
-// Create
+// Create listing
 listingRouter.post('/', passport.authenticate('jwt', { session: false }), async (req, res) => {
-	try{
+	try {
+		let {
+			skill,
+			serviceDetails,
+			suggestedPrice,
+			location
+		} = req.body;
+		let listing = await Listing.create({
+			skill,
+			isActive: true,
+			serviceDetails,
+			suggestedPrice,
+			location
+		})
+		res.json({ msg: `Listing ${listing.skill} created.` })
+	} catch (e) {
+		res.json({ Error: `${e}` });
+	}
+});
 
-		res.json({msg: ``})
-	}catch (e){
-		res.json({Error: `${e}`});
+// Read all listings
+listingRouter.get('/', passport.authenticate('jwt', { session: false }), async (req, res) => {
+	try {
+		let listings = await Listing.findAll();
+		res.json({ listings })
+	} catch (e) {
+		res.json({ Error: `${e}` });
 	}
 });
 
-// Read
+// Read individual listing
 listingRouter.get('/:id', passport.authenticate('jwt', { session: false }), async (req, res) => {
-	try{
-
-		res.json({msg: ``})
-	}catch (e){
-		res.json({Error: `${e}`});
+	try {
+		let listing = await Listing.findByPk(req.params.id);
+		res.json({ listing })
+	} catch (e) {
+		res.json({ Error: `${e}` });
 	}
 });
 
-// Update
+// Update listing
 listingRouter.put('/:id', passport.authenticate('jwt', { session: false }), async (req, res) => {
-	try{
-
-		res.json({msg: ``})
-	}catch (e){
-		res.json({Error: `${e}`});
+	try {
+		let listing = await Listing.findByPk(req.params.id);
+		let { skill, isActive, serviceDetails, suggestedPrice, location } = req.body;
+		listing.update({
+			skill,
+			isActive,
+			serviceDetails,
+			suggestedPrice,
+			location
+		});
+		listing.save();
+		res.json({ msg: `Listing ${listing.skill} updated` });
```

Comment:
This is great, but now that I'm looking at it, we want to make sure that a user may only update her own listing. Other wise any user could update anything they wanted.

We don't have relationships set up yet, but eventually a User with have many Listings, and a Listing will belong to one User. 

```
                let listing = await Listing.findByPk(req.params.id);
                const { user } = req; 
                const userListings = user.getListings();
                if (userListings.includes(listing.id)) {
		    let { skill, isActive, serviceDetails, suggestedPrice, location } = req.body;
		    listing.update({
			skill,
			isActive,
			serviceDetails,
			suggestedPrice,
			location
		    });
		    listing.save();
		    res.json({ msg: `Listing ${listing.skill} updated` });
                }
```

After the associations are made, this route could look something like this. For now, let's just leave a comment that we need to update this route later with specific user protection.